codegra_fs/cgfs.py

```diff
@@ -1621,9 +1907,109 @@ def write(self, path, data, offset, fh):
             return file.write(data, offset)
 
 
-def main():
+def create_and_mount_fs(
+    username: str,
+    password: str,
+    url: t.Optional[str],
+    fixed: bool,
+    assigned_only: bool,
+    latest_only: bool,
+    mountpoint: str,
+    rubric_append_only: bool,
+) -> None:
+    global cgapi
+    logger.info('Mounting... ')
+
+    cgapi = CGAPI(
+        username,
+        password,
+        url,
+        fixed=fixed,
+    )
+
+    if not fixed:
+        logger.warning('=====================================================')
+        logger.warning('Mounting in non-fixed mode, all changes will be')
+        logger.warning('visible and additions to students.')
+        logger.warning('Watch out when uploading grading scripts!')
+        logger.warning('=====================================================')
+
+    with tempfile.TemporaryDirectory(dir=tempfile.gettempdir()) as tmpdir:
+        sockfile = tempfile.NamedTemporaryFile().name
+        kwargs: t.Dict[str, str] = {}
+        if sys.platform.startswith('darwin'):
+            # Fix OSX encoding issue as described here:
+            # https://web.archive.org/web/20180920131107/https://github.com/osxfuse/osxfuse/issues/71
+            kwargs = {
+                'from_code': 'UTF-8',
+                'to_code': 'UTF-8-MAC',
+                'modules': 'iconv',
+            }
+
+        fs = None
+        try:
+            fs = CGFS(
+                latest_only=latest_only,
+                socketfile=sockfile,
+                fixed=fixed,
+                mountpoint=mountpoint,
+                tmpdir=tmpdir,
+                rubric_append_only=rubric_append_only,
+                assigned_only=assigned_only,
+            )
+            FUSE(
+                fs,
+                mountpoint,
+                nothreads=True,
+                foreground=True,
+                direct_io=True,
+                **kwargs,
+            )
+        except RuntimeError:  # pragma: no cover
+            logger.critical(traceback.format_exc())
+        finally:
+            if fs is not None and hasattr(fs, 'api_handler'):
+                fs.api_handler.stop = True
+            if os.path.isfile(sockfile):
+                os.unlink(sockfile)
+
+
+def newer_version_available() -> bool:
+    req = requests.get('https://codegra.de/.cgfs.version', timeout=2)
+    return req.status_code < 300 and tuple(
+        int(p) for p in req.content.decode('utf8').strip().split('.')
+    ) > codegra_fs.__version__
+
+
+def check_version() -> None:
+    if newer_version_available():
+        msg = [
+            (
+                'You are running an outdated version of'
+                ' CGFS, please consider upgrading.'
+            ),
+            'You can do this at https://codegra.de/codegra_fs/latest/',
```

Comment:
This URL gives a 404. You probably already know that, just mentioning it so we don't forget about it :)

Reply:
Yeah, we need to add it when we actually release something :)